### PR TITLE
Remove throttle, use `IDistributedCache`

### PR DESCRIPTION
**Pull request by emanuelgaspar**
The `CoinGecko` client had a throttle to ensure we don't go over the maximum allowed calls per second/minute.

The previous throttle has a few issues:
 - the `SemaphoreSlim` lock only allows 1 call at a time. Reduces concurrency.
 - cache responses in `IMemoryCache` to avoid duplicate calls. This only works for a single service.

This PR adds a few drastic but necessary changes.

**Throttle has been removed.**
The throttle was probably necessary when we were using the free/public version.
With our current "Pro" subscription, we can run ~16 calls per second.
If we eventually do need a throttle, its implementation needs to change anyway, to allow greater concurrency.

**`IDistributedCache` instead of `IMemoryCache`**
If we really want to avoid duplicate calls, we should reuse responses across all services that need a connection to `CoinGecko`, not individually for each service. In that regard, saving all responses in `Redis` is more robust and should help save even more calls to the external API.

### CODE REVIEW

File: src/Trakx.CoinGecko.ApiClient/Clients/CachedHttpClientHandler.cs [hunk 1]
```diff
@@ -1,100 +1,69 @@
-﻿using System;
-using System.Linq;
-using System.Net;
+﻿using System.Net;
```
Comment (emanuelgaspar) [inline src/Trakx.CoinGecko.ApiClient/Clients/CachedHttpClientHandler.cs:1]: I recommend reviewing this file in full:
<https://github.com/trakx/coingecko-api-client/blob/837721011c4414ac3f4c6d05854f3ea3e8092525/src/Trakx.CoinGecko.ApiClient/Clients/CachedHttpClientHandler.cs>

It's too different from the previous version, so the comparison is weird.

File: src/Trakx.CoinGecko.ApiClient/Configuration/ApiClientExtensions.cs [hunk 2]
```diff
@@ -20,27 +21,28 @@ public static partial class ApiClientExtensions
...
+        var apiConfiguration = configuration.GetConfiguration<CoinGeckoApiConfiguration>();
+        var cacheConfiguration = configuration.GetConfiguration<RedisCacheConfiguration>();
+        return services.AddCoinGeckoClient(apiConfiguration, cacheConfiguration);
     }
 
     public static IServiceCollection AddCoinGeckoClient(
-        this IServiceCollection services, CoinGeckoApiConfiguration apiConfiguration)
+        this IServiceCollection services,
+        CoinGeckoApiConfiguration apiConfiguration,
+        RedisCacheConfiguration cacheConfiguration)
```
Comment (emanuelgaspar) [inline src/Trakx.CoinGecko.ApiClient/Configuration/ApiClientExtensions.cs:32]: Adding `RedisCacheConfiguration` here is a breaking change.

File: src/Trakx.CoinGecko.ApiClient/Configuration/ApiClientExtensions.cs [hunk 3]
```diff
@@ -20,27 +21,28 @@ public static partial class ApiClientExtensions
...
-
-        services.AddSingleton<ICoinGeckoClient, CoinGeckoClient>();
-        services.AddTransient<CachedHttpClientHandler>();
+        services.AddSingleton(new ClientConfigurator(apiConfiguration));
 
+        // required common services
         services.AddSingleton<IDateTimeProvider, DateTimeProvider>();
-        services.AddSingleton<ISemaphore>(new Semaphore(new SemaphoreSlim(1, 1)));
+        services.AddDistributedCache(cacheConfiguration);
         services.AddMemoryCache();
```
Comment (emanuelgaspar) [inline src/Trakx.CoinGecko.ApiClient/Configuration/ApiClientExtensions.cs:41]: I'm not overly fond of setting up these "external" services here, but I understand how it makes it easier to consume this api client.

**Review by monsieurleberre — APPROVED** [review]
This PR looks good :)
But then outside of the scope of the PR I wonder :
- Should we really cache all responses, when some are supposed to give most up to date price, market cap, etc. Maybe it would be safer to cache the historical stuff for while, but almost never the "latest" endpoints response?
- Should we look at using an existing contract from dotnet like [HttpCacheRequestPolicy](https://learn.microsoft.com/en-us/dotnet/api/system.net.cache.httprequestcachepolicy?view=net-7.0)?

**Comment by emanuelgaspar** [comment]
> This PR looks good :) But then outside of the scope of the PR I wonder...

Very good points. Created a `tech debt` task for them:
https://trakx.atlassian.net/browse/TKX-1146

**Pull request merged**